### PR TITLE
mzcompose: Support CI_EXTRA_ARGS

### PR DESCRIPTION
**Pull request by def-**
Tested locally with `CI_EXTRA_ARGS='{"feature-benchmark": "--scenario=Foasdb"}' BUILDKITE_STEP_KEY=feature-benchmark bin/mzcompose --find feature-benchmark run default`
Looks like this in trigger-ci: 
Screenshot 2024-07-18 at 01 02 26
Code: https://github.com/MaterializeInc/trigger-ci/commit/eb9c9f686a1bddd8a56f3575f9fdb2eb5dc3ab09 (not live yet)

Proof that this works: https://buildkite.com/materialize/nightly/builds/8639
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: misc/python/materialize/cli/mzcompose.py [hunk 1]
```diff
@@ -625,7 +626,16 @@ def handle_composition(
             # test analytics, even if the workflow doesn't define more granular
             # test cases.
             with composition.test_case(f"workflow-{args.workflow}"):
-                composition.workflow(args.workflow, *args.unknown_subargs[1:])
+                ci_extra_args = json.loads(os.getenv("CI_EXTRA_ARGS", "{}"))
+                buildkite_step_key = os.getenv("BUILDKITE_STEP_KEY")
+                extra_args = (
+                    ci_extra_args[buildkite_step_key].split(" ")
```
Comment (nrainer-materialize) [inline misc/python/materialize/cli/mzcompose.py:632]: Why do you split at the space? Should it be the colon?

File: misc/python/materialize/cli/mzcompose.py [hunk 2]
```diff
@@ -625,7 +626,16 @@ def handle_composition(
...
+                    if buildkite_step_key
+                    else []
+                )
+                composition.workflow(
+                    args.workflow, *args.unknown_subargs[1:], *extra_args
```
Comment (nrainer-materialize) [inline misc/python/materialize/cli/mzcompose.py:637]: Do the `extra_args` override existing args if they are already set?

Comment (def-) [reply misc/python/materialize/cli/mzcompose.py:637]: Depends on how Python interprets them. We just append them after the normal args. You can use argparse to overwrite with the last value, or interpret them as a list.

File: misc/python/materialize/cli/mzcompose.py (hunk 1 above)
Comment (def-) [reply misc/python/materialize/cli/mzcompose.py:632]: Seems more natural to list the arguments the same way as you'd do on command line.

Comment (nrainer-materialize) [reply misc/python/materialize/cli/mzcompose.py:632]: Ah, got it. This is the part we are applying it to: ` "--scenario=Foasdb"`

Comment (def-) [reply misc/python/materialize/cli/mzcompose.py:632]: Actually shlex.split is even better: https://docs.python.org/3/library/shlex.html#shlex.split will use that.

File: misc/python/materialize/cli/mzcompose.py (hunk 2 above)
Comment (nrainer-materialize) [reply misc/python/materialize/cli/mzcompose.py:638]: Should we use `setattr(args, param, new_value)`?

**Review by nrainer-materialize — APPROVED** [review]

**Review by nrainer-materialize — APPROVED** [review]

**Review by nrainer-materialize — APPROVED** [review]

**Pull request merged**